tests/chains/Polkadot/TWAnySignerTests.cpp

```diff
@@ -87,18 +130,20 @@ TEST(PolkadotSigner, SignTransferDOT) {
     auto balanceCall = input.mutable_balance_call();
     auto& transfer = *balanceCall->mutable_transfer();
     auto value = store(uint256_t(12345));
-    transfer.set_to_address(toAddress.string());
+    transfer.set_to_address(TWStringUTF8Bytes(toAddressStr.get()));
     transfer.set_value(value.data(), value.size());
 
-    auto extrinsic = Extrinsic(input);
-    auto preimage = extrinsic.encodePayload();
-    auto output = Signer::sign(input);
+    auto preimage = helper_encodePayload(TWCoinTypePolkadot, input);
+    EXPECT_EQ(hex(preimage), "050088dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0eee5c032000000110000000300000091b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3343a3f4258fd92f5ca6ca5abdf473d86a78b0bcd0dc09c568ca594245cc8c642");
+
+    Proto::SigningOutput output;
+    ANY_SIGN(input, TWCoinTypePolkadot);
+    EXPECT_EQ(output.error(), Common::Proto::OK);
 
-    ASSERT_EQ(hex(preimage), "05008eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48e5c032000000110000000300000091b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3343a3f4258fd92f5ca6ca5abdf473d86a78b0bcd0dc09c568ca594245cc8c642");
-    ASSERT_EQ(hex(output.encoded()), "29028488dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee003d91a06263956d8ce3ce5c55455baefff299d9cb2bb3f76866b6828ee4083770b6c03b05d7b6eb510ac78d047002c1fe5c6ee4b37c9c5a8b09ea07677f12e50d3200000005008eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48e5c0");
+    ASSERT_EQ(hex(output.encoded()), "29028488dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee0061ee605d8cdcedf93200ba01636572f032ff04d75ed2999b80c184bc03ecbca7df83d176d955551b8f1654d2f6f771565be8d573df82d0b7f2eaba6d5daa3f0232000000050088dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0eee5c0");
```

Comment:
Why these values have been changed? It's important to keep backward compatibility

Reply:
Not sure why I changed it originally.  But I have got it working with the original expected outputs.  The previous test was decoding hex to an ed25519 public key, but the Rust `PublicKey` was rejecting it (maybe it was a valid sr25519 key?).  It seems the C++ public key code doesn't do the "decompress curve" test that the Rust code does.

Since the value was only used as the `to_address` in a transfer, we didn't need to require it to be a valid ed25519 key, so I have convert the public key to a ss58 address.